src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeMetricsCollectorSpark.java

```diff
@@ -0,0 +1,156 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.util.Histogram;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SamPairUtil;
+import htsjdk.samtools.metrics.MetricsFile;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+import org.broadinstitute.hellbender.tools.picard.analysis.InsertSizeMetrics;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Worker class to collect insert size metrics.
+ */
+public final class InsertSizeMetricsCollectorSpark implements Serializable{
+    private static final long serialVersionUID = 1L;
+
+    private final JavaRDD<Integer> lengthRDD;
+    private final InsertSizeMetrics metric = new InsertSizeMetrics();
+    final Histogram<Integer> htslibHistogram;
+
+    /**
+     * Constructor and perform map & reduce.
+     * @param filteredReads          reads that pass filters
+     * @param samRgRecords           a list of read groups
+     * @param HistogramMADTolerance  MAD tolerance when producing histogram plot
+     */
+    // TODO: * @param accumLevels            accumulation level {SAMPLE, LIBRARY, READGROUP}
+    public InsertSizeMetricsCollectorSpark(final JavaRDD<GATKRead> filteredReads,
+                                           final long pairsCount,
+                                           //final Set<MetricAccumulationLevel> accumLevels,
+                                           final List<SAMReadGroupRecord> samRgRecords,
+                                           final double HistogramMADTolerance) {
+
+        lengthRDD = filteredReads.map(read -> Math.abs(read.getFragmentLength()) );
+
+        double lengthArr[] = returnRDDArray(lengthRDD);
+
+        DescriptiveStatistics percent = new DescriptiveStatistics(lengthArr);
```

Comment:
this code here should be extracted to its own method. This constructor is way too long